src/actions/actions.ts

```diff
@@ -2694,3 +2695,76 @@ class ToggleCaseAndMoveForward extends BaseMovement {
     return position.getRight();
   }
 }
+
+abstract class IncrementDecrementNumberAction extends BaseMovement {
+  modes = [ModeName.Normal];
+  canBePrefixedWithCount = true;
+
+  offset: number;
+
+  public async execActionWithCount(position: Position, vimState: VimState, count: number): Promise<Position> {
+    count = count || 1;
+    const text = TextEditor.getLineAt(position).text;
+
+    // They may start on a number, so check the current word boundaries first.
+    let wordEnd = position.getCurrentWordEnd(true);
+    let wordStart = position.getWordLeft(true);
+    do {
```

Comment:
Would you mind abstracting out a function in the Position class called something like IterateWords in the same style of IterateDocument and IterateLineStart?